tools/KoreBuild.Console/Commands/InstallToolsetsCommand.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.CommandLineUtils;
+
+namespace KoreBuild.Console.Commands
+{
+    internal class InstallToolsetsCommand : SubCommandBase
+    {
+        private CommandOption _quietOpt;
+        private CommandOption _upgradeOpt;
+
+        public InstallToolsetsCommand(CommandContext context) : base(context)
+        {
+        }
+
+        public override void Configure(CommandLineApplication application)
+        {
+            application.Description = "Installs the toolsets necessary to build the current project.";
+            application.ExtendedHelpText = @"
+MORE INFO:
+
+    Uses the toolsets specified in korebuild.json to install all toolsets.
+";
+            _quietOpt = application.Option("-q|--quiet",
+                "Install toolsets without requiring user interation.",
+                CommandOptionType.NoValue);
+            _upgradeOpt = application.Option("-u|--upgrade",
+                "Upgrade existing toolsets.",
+                CommandOptionType.NoValue);
+            base.Configure(application);
+        }
+
+        protected override int Execute()
+        {
+            var args = new List<string>
```

Comment:
Any particular reason you chose to implement this as a two-process hop into MSBuild? Or just because this is what the other tasks are doing?

It seems like we don't really need really need MSBuild to run this command, so IMO this would benefit from the simplicity of running the vs installer directly from KoreBuild.Console.

Reply:
Few reasons:
- Code sharing. We need to parse the korebuild.json file, find the vs installation (vs_where), downloading the vs file, etc were all part of the KoreBuild.Tasks project already. I didn't see a great way to share code between the Korebuild.Console and Korebuild.Tasks projects, so I found implement it this way was easier than moving everything around.
- Extensibility. Implementing things in Korebuild.Console seems to become more difficult the more expansive it gets. For example, if we wanted to install toolsets for nodejs, we would need to move even more things to a shared project.

I originally started with having everything in Korebuild.Console, but I quickly realized that if I wanted to parse the korebuild.json file and find vs, it would be simpler to move everything into Korebuild.Tasks.

Reply:
Ok, that's good enough justification for me.